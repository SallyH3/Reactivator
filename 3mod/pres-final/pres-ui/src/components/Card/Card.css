.card-holder {
  border: 3px solid white;
  margin: 20px;
  padding: 20px;
  height: 250px;
  width: 200px;
}