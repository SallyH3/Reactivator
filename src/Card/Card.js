import React, { Component } from 'react';
import data from '../data.js';

class Card extends Component {
  constructor(props) {
    super(props);
    this.state = {
      input: '',
      message: ''
    }
  }

  checkAnswer = () => {
    if(this.state.input === this.props.answer) {
      this.setState({
        message: 'Correct - nice work!'
      })
    } else {
      this.setState({
        message: 'Almost! Try again!'
      })
      
    }
  }
  
  handleReturn = () => {
    this.setState({input: ''})
    this.checkAnswer();
  }

  handleInput = (e) => {
    this.setState({
      input: e.target.value
    })
  }

  handleInputChange = (e) => {
    e.preventDefault()
    //fix below so that user can enter upper or lowercase and it still checks answer correctly
    let userInput = this.state.input.toLowerCase();
    let result = this.checkAnswer(userInput);
    this.handleReturn(result)
  }


  render() {
    return (
      <main className='card-container'>
        <h3 className='subject'>
          Subject: {this.props.subject}
        </h3>
        <h4 className='question'>
          Question: {this.props.question}
        </h4>
        <form 
          className='form'
          onSubmit={this.handleInputChange}>
          <input 
            type='text'
            value={this.state.input}
            onChange={this.handleInput}
            placeholder="Enter answer and press return"
            className='user-answer'>
          </input>
          <p className = 'message'>{this.state.message}</p>
        </form>
      </main>
    )
  }
}

export default Card;