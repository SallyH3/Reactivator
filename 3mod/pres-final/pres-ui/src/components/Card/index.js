import React from 'react';
import './Card.css';

export const Card = (props) => {
  const {number, president, birthYear, deathYear, tookOffice, leftOffice, party} = props

  return(
    <div className='card-holder'>
      <p>{number}</p>
      <p>{president}</p>
      <p>{birthYear}</p>
      <p>{deathYear}</p>
      <p>{tookOffice}</p>
      <p>{leftOffice}</p>
      <p>{party}</p>
    </div>
  )
}

export default Card;