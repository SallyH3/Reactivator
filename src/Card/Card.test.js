import React from 'react';
import Card from './Card';
import { shallow } from 'enzyme';

jest.useFakeTimers();

let mockAnswer = 'react elements';

const mockMsg = 'Correct - nice work!';

let mockData = [
  {
    subject: "JSX",
    question: "JSX represents objects called what?",
    answer: "React elements",
    correct: false,
    id: 1
  },
  {
    subject: "JSX",
    question: "JSX is a combination of these two languages",
    answer: "HTML and JavaScript",
    correct: false,
    id: 2
  }
];

describe ('Card', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Card card={mockData}
      answer={mockAnswer}
      message={mockMsg}
      />
    )
  });

  it('should have default states', () => {
    expect(wrapper.state()).toEqual({
      input: '',
      message: ''
    });
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should check answer from user', () => {
    wrapper.find('.user-answer').simulate('change', {target: {value: 'React elements'}});
    expect(mockAnswer).toEqual('react elements');
  });

  it('should check if setTimeout has been called', () => {
    wrapper.find('.form').simulate('submit', { preventDefault: () => {}});
    expect(setTimeout).toHaveBeenCalledTimes(1);
  });

  it('should setState with a message to user if they answered incorrectly', () => {
    expect(wrapper.state('message')).toEqual('')
    wrapper.instance().checkAnswer();
    wrapper.find('.form').simulate('keypress', {key: 'Enter'})
    expect(wrapper.state('message')).toEqual('Not quite!');
  });

  it('should setState to empty string after user presses return', () => {
    wrapper.instance().handleReturn();
    expect(wrapper.state().input).toEqual('');
  });
});