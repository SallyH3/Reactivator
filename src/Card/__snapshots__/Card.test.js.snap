// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
<main
  className="card-container"
>
  <section
    className="cards"
  >
    <h3
      className="subject"
    >
      Subject: 
    </h3>
    <h4
      className="question"
    >
      Question: 
    </h4>
    <form
      className="form"
      onSubmit={[Function]}
    >
      <input
        className="user-answer"
        onChange={[Function]}
        placeholder="Enter answer and press return"
        type="text"
        value=""
      />
      <p
        className="message"
      />
    </form>
  </section>
</main>
`;
